homeassistant/util/yaml.py

```diff
@@ -31,11 +31,17 @@ class NodeStrClass(str):
         """Wrapper class to be able to add attributes on a string."""
 
         pass
+    
+    class NodeDictClass(dict):
+        """Wrapper class to be able to add attributes on a dict."""
 
+        pass
     if isinstance(obj, list):
         obj = NodeListClass(obj)
     if isinstance(obj, str):
         obj = NodeStrClass(obj)
+    if isinstance(obj, dict):
```

Comment:
I agree with both. Also would be interested to see where non ordered dicts come from. 

Looks like when we include files, we don't convert to ordereddict anymore?

```
  File "/srv/homeassistant/homeassistant_venv/lib/python3.4/site-packages/homeassistant/util/yaml.py", line 94, in _include_yaml
    return _add_reference(load_yaml(fname), loader, node)
  File "/srv/homeassistant/homeassistant_venv/lib/python3.4/site-packages/homeassistant/util/yaml.py", line 39, in _add_reference
    setattr(obj, '__config_file__', loader.name)
AttributeError: 'dict' object has no attribute '__config_file__'
```

_(stacktrace from #5354)_